### PR TITLE
search for skills improved

### PR DESCRIPTION
**Pull request by SimonLab**
200 :sparkles:
## Before merging this PR:
we should close this [issue 201](https://github.com/FAC-GM/app/issues/201) which is linked to this [PR on search-result-keyword-highlighter](https://github.com/dwyl/search-result-keyword-highlighter/pull/16)

## Then
[issue 193](https://github.com/FAC-GM/app/issues/193) Search with special character (eg: c++) now return relevant data.

@nelsonic can you have a look at this PR? Thanks

### CODE REVIEW

File: lib/query.js [hunk 1]
```diff
@@ -31,8 +31,9 @@ exports.register = function(server, options, next) {
           var skills = request.query.skills.split(',');
           skills = skills.map(deleteSpaces);
           skills = skills.filter(a => a !== '');
-          skills = skills.map(escapeValue);
```
Comment (SimonLab) [inline lib/query.js]: Testing with special character + it looks like the elasticsaerch query doesn't need to escape the character. that's why I delete this line

**Comment by SimonLab** [comment]
The main work was done by reindexing the data with the right analyzer: Instead of creating new token from a value (c++, c# are for example transformed to only c) the new analyzer create a token which is the same as the value (c++ will stay c++)

File: lib/query.js [hunk 2]
```diff
@@ -31,8 +31,9 @@ exports.register = function(server, options, next) {
...
-
+          skills = skills.map(function(skill) {
+            return skill.toLowerCase();
```
Comment (SimonLab) [inline lib/query.js]: The skills name are now saved in lower case into elasticasearch so our search need to be in lower case too to match.

**Comment by SimonLab** [comment]
The highlight of special characters is now updated, see [issue 201](https://github.com/FAC-GM/app/issues/201)

**Comment by nelsonic** [comment]
Nice one @SimonLab :+1:

**Pull request merged**